RecoLocalMuon/DTRecHit/plugins/DTLinearDriftFromDBAlgo.cc

```diff
@@ -34,19 +36,37 @@ DTLinearDriftFromDBAlgo::DTLinearDriftFromDBAlgo(const ParameterSet& config)
       // Option to force going back to digi time at Step 2
       stepTwoFromDigi(config.getParameter<bool>("stepTwoFromDigi")),
       useUncertDB(config.getParameter<bool>("useUncertDB")),
+      readLegacyTTrigDB(config.existsAs<bool>("readLegacyTTrigDB") ? config.getParameter<bool>("readLegacyTTrigDB")
+                                                                   : true),
+      readLegacyVDriftDB(config.existsAs<bool>("readLegacyVDriftDB") ? config.getParameter<bool>("readLegacyVDriftDB")
+                                                                     : true),
       // Set verbose output
       debug(config.getUntrackedParameter<bool>("debug")) {}
 
 DTLinearDriftFromDBAlgo::~DTLinearDriftFromDBAlgo() {}
 
 void DTLinearDriftFromDBAlgo::setES(const EventSetup& setup) {
   if (debug)
-    cout << "[DTLinearDriftFromDBAlgo] setES called" << endl;
+    edm::LogVerbatim("DTLocalReco") << "[DTLinearDriftFromDBAlgo] setES called" << endl;
   theSync->setES(setup);
   // Get the map of ttrig from the Setup
-  ESHandle<DTMtime> mTimeHandle;
-  setup.get<DTMtimeRcd>().get(mTimeHandle);
-  mTimeMap = &*mTimeHandle;
+  if (readLegacyVDriftDB) {
+    ESHandle<DTMtime> mTimeHandle;
+    setup.get<DTMtimeRcd>().get(mTimeHandle);
+    mTimeMap = &*mTimeHandle;
+    vDriftMap = nullptr;
+  } else {
+    ESHandle<DTRecoConditions> hVdrift;
+    setup.get<DTRecoConditionsVdriftRcd>().get(hVdrift);
```

Comment:
this additional `es.get` goes against the current campaign to define esConsumes.
Please consider adding esConsumes in this PR already.

Reply:
OK will have a look on this on Monday (other get's will have to be migrated in the same class as well)

Reply:
Uhm, this class is an helper that is called by the actual EDProducer, but is not an EDProducer itself.
So I cannot directly call esConsumes from here. ("error: 'esConsumes' was not declared in this scope")
@slava77, do you know how should I handle this use case? I checked the doc but found nothing apparently related.

Reply:
the consumesCollector will need to be passed to the algorithm constructor.
a (somewhat random) example is available in isolation 
https://github.com/cms-sw/cmssw/blob/49cd0eb9a47e85d38023c5c158e60be18687ee04/RecoMuon/MuonIsolationProducers/plugins/MuIsoDepositProducer.cc#L74

Reply:
Mmhh this would mean updating every implementation under [RecoLocalMuon/DTRecHit/plugins](https://github.com/cms-sw/cmssw/tree/master/RecoLocalMuon/DTRecHit/plugins), don't you think it would make sense as a separate migration rather than being coupled to this specific PR?

Reply:
it's OK to do in a follow up PR.

Reply:
One thing I could do is to add this in the algo interface with cc=0 as a default and leave it dummy in other implementations not affected by this PR. Do you like the idea?

Reply:
without compiling this locally, I'm not sure that this would help much. 

We did something along the lines in the MuonServiceProxy; there the solution was to add an interface (constructor and related calls) with `ConsumesCollector`. But then in this case it will probably just complicate the implementation with something artificially temporary.